plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/variation-selector/frontend.ts

```diff
@@ -36,13 +36,101 @@ type Context = AddToCartWithOptionsStoreContext & {
 };
 
+// When true, do not handle clicks on attributes, this is to avoid infinite recursion
+let inAutoselectScope = false;
+
 // Set selected pill styles for proper contrast.
 setStyles();
 
 // Stores are locked to prevent 3PD usage until the API is stable.
 const universalLock =
 	'I acknowledge that using a private store means my plugin will inevitably break on the next store release.';
 
+function attributesAutoselect( variation_selectors ) {
+	const { selectedAttributes } =
+		getContext< Context >();
+			// 'pill' is not actually a possible optionStyle value, pills are simply the default option, and they don't give a value to optionStyle
+	for ( const current_variation_selector of variation_selectors ) {
+		const optionStyle = current_variation_selector.dataset?.optionStyle || 'pills';
+		// Dropdown options or Pill inputs,
+		// that HAVE a value (Choose an Option has an empty value of ""),
+		// and are compatible with the possible variations
+		let valid_choices = [];
+		current_variation_selector.querySelectorAll( 'option:not([value=""]), input.wc-block-add-to-cart-with-options-variation-selector-attribute-options__pill-input:not([value=""])' ).forEach( ( el ) => {
+			var name = '';
+			switch ( optionStyle ) {
+				case 'pills':
+					name = el.getAttribute( 'name' );
+					break;
+				case 'dropdown':
+					name = JSON.parse( el.dataset?.wpContext )[ 'name' ];
+					break;
+				default:
+					throw new Error( 'optionStyle not implemented!: ' + optionStyle );
+			}
+			if ( isAttributeValueValid( {
+				attributeName: name,
+				attributeValue: el.value,
+				selectedAttributes,
+			} ) ) {
+				valid_choices.push( el );
+			}
+		} );
+		if ( valid_choices.length === 1 ) {
+			// Only 1 option (+ the "Choose an option" choice in case of dropdowns)
+			const valid_choice = valid_choices[0];
+			const selected = current_variation_selector.querySelectorAll( ':checked' );
+			if ( selected.length === 0 || selected[0].value !== valid_choice.value ) {
+				// No option selected, OR the selected value is not the same as the only valid one (for example if the selected value is "" (Choose an Option))
+				inAutoselectScope = true;
+				let ev;
+				switch ( optionStyle ) {
+					case 'pills':
+						// Manually enable the input, because we know it is valid and will be enabled by Wordpress's interactivity API anyways.
+						valid_choice.classList.remove( 'disabled' )
+						valid_choice.removeAttribute( 'disabled' );
+						ev = new MouseEvent( 'click', { bubbles: true } );
+						valid_choice.dispatchEvent( ev );
+						break;
+					case 'dropdown':
+						const select = valid_choice.parentElement;
+						select.value = valid_choice.value;
+						ev = new Event( 'change' , { bubbles: true } );
+						select.dispatchEvent( ev );
+						break;
+					default:
+						throw new Error( 'optionStyle not implemented!: ' + optionStyle );
+				}
+				inAutoselectScope = false;
+			}
+		}
+	}
+}
+
+function attributesAutoselectOthers() {
```

Comment:
Same here, we could change this from `attributesAutoselectOthers()` to `autoselectOtherAttributes()`.